### ISSUE TITLE
There is no documentation on how to suppress warning annotations that occur in files not modified in a given pull request

### ISSUE THREAD

**Issue by JasonGross** [actions/toolkit#559]
The document at https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md makes no mention of how to suppress warnings that occur in files not modified in a given PR.  If this feature is not yet available, it should be made available (either on a per-matcher or, perhaps, a per-repository basis) and there should be documentation about how to accomplish this.

**Comment by electronjoe** [comment]
Agree, I have a [PR](https://github.com/actions/toolkit/pull/734) up for the documentation to make Problem Matcher docs more clear on the annotation limits today (which combine poorly with lack of filtering, as you call out). I have 600 annotations in our project - from GCC warnings. I ultimately added shell processing of the tool warnings to pre-filter by files modified (e.g. using get-changed-files@v2 - which itself has some issues with workflows that involve forked-branch-PR-to-upstream-master). If a tool only allows 10 annotations per step, some form of filtering support is pretty important (or everybody will roll their own).
